hopr/hopr-lib/src/helpers.rs

```diff
@@ -0,0 +1,46 @@
+use std::sync::Arc;
+
+use async_lock::RwLock;
+use hopr_internal_types::protocol::TagBloomFilter;
+use hopr_platform::file::native::{read_file, write};
+use tracing::{debug, error, info};
+
+#[derive(Debug, Clone)]
+pub(crate) struct WrappedTagBloomFilter {
+    path: String,
+    tbf: Arc<RwLock<TagBloomFilter>>,
+}
+
+impl WrappedTagBloomFilter {
+    pub fn new(path: String) -> Self {
+        info!("Creating the Bloom filter storage at: {}", path);
+        let tbf = read_file(&path)
+            .and_then(|data| {
+                TagBloomFilter::from_bytes(&data)
+                    .map_err(|e| hopr_platform::error::PlatformError::GeneralError(e.to_string()))
+            })
+            .unwrap_or_else(|_| {
+                debug!("No tag Bloom filter found, using empty");
+                TagBloomFilter::default()
+            });
+
+        Self {
+            path,
+            tbf: Arc::new(RwLock::new(tbf)),
+        }
+    }
+
+    pub fn raw_filter(&self) -> Arc<RwLock<TagBloomFilter>> {
```

Comment:
Should be maybe called `inner()`

Reply:
I'd prefer for this to have a `with_inner` and execute something in that context.